code/modules/projectiles/guns/energy/blaster.dm

```diff
@@ -115,4 +115,67 @@
 	if(wielded)
 		toggle_scope(2.0, usr)
 	else
-		to_chat(usr, "<span class='warning'>You can't look through the scope without stabilizing the rifle!</span>")
\ No newline at end of file
+		to_chat(usr, "<span class='warning'>You can't look through the scope without stabilizing the rifle!</span>")
+
+/obj/item/gun/energy/secblaster
+	name = "service blaster"
+	desc = "The NT BP-7 is a multi-mode blaster pistol developed and produced by Nanotrasen for its internal security departments." // Placeholder for the PR. If there's a loreman who wants to describe this monstrosity, just hmu
+	icon = 'icons/obj/guns/secblaster/secblasters.dmi'
+	icon_state = "secblaster"
+	item_state = "secblaster"
+	fire_sound = 'sound/weapons/secblasterstun.ogg'
+	slot_flags = SLOT_BELT|SLOT_HOLSTER
+	w_class = 2
+	force = 5
+	origin_tech = list(TECH_COMBAT = 3, TECH_MAGNET = 2)
+	matter = list(DEFAULT_WALL_MATERIAL = 2000)
+	projectile_type = /obj/item/projectile/energy/stunblaster
+	secondary_projectile_type = /obj/item/projectile/energy/blaster
+	max_shots = 12 //12 shots stun, 8 shots lethal.
+	charge_cost = 50
+	has_item_ratio = FALSE
+	modifystate = "secblasterstun"
+	var/modelselected = FALSE
+	sel_mode = 1
+	firemodes = list(
+		list(mode_name="stun", projectile_type=/obj/item/projectile/energy/stunblaster, modifystate="secblasterstun", charge_cost = 50, fire_sound = 'sound/weapons/secblasterstun.ogg'),
+		list(mode_name="lethal", projectile_type=/obj/item/projectile/energy/blaster, modifystate="secblasterkill", recoil = 1, charge_cost = 75, fire_sound = 'sound/weapons/secblasterlethal.ogg')
+		)
+
+/obj/item/gun/energy/secblaster/verb/select_frame()
```

Comment:
This verb should remove itself when it completes successfully.

Reply:
Sorry, I don't know how to do that. Could you explain it to me and I'll see about making it so.

Reply:
After you successfully select the frame, do: 

verbs -= /obj/item/gun/energy/secblaster/verb/select_frame

Reply:
Champion! thank you.

Reply:
Donezo.